js/plugins/iq.js

```diff
@@ -0,0 +1,348 @@
+require("../framework/InitAnalyticsNamespace.js");
+require("../../html5-common/js/utils/utils.js");
+
+/**
+ * @class IqPlugin
+ * @classdesc This is an example class of a plugin that works with the Ooyala Analytics Framework.
+ * @param {object} framework The Analytics Framework instance
+ */
+var IqPlugin= function (framework)
+{
+  var _framework = framework;
+  var name = "iq";
+  var version = "v1";
+  var id;
+
+  var SDK_LOAD_TIMEOUT = 3000;
+
+  var autoPlay = null;
+  var pcode = null;
+  var playerId = null;
+  var currentEmbedCode = null;
+  var contentType = "ooyala";
+  var currentPlayheadPosition = null;
+  
+  this.ooyalaReporter = null;
+  this.testMode = false;
+
+  /**
+   * [Required Function] Return the name of the plugin.
+   * @public
+   * @method IqPlugin#getName
+   * @return {string} The name of the plugin.
+   */
+  this.getName = function ()
+  {
+    return name;
+  };
+
+  /**
+   * [Required Function] Return the version string of the plugin.
+   * @public
+   * @method IqPlugin#getVersion
+   * @return {string} The version of the plugin.
+   */
+  this.getVersion = function ()
+  {
+    return version;
+  };
+
+  /**
+   * Return the autoPlay value.
+   * @public
+   * @method IqPlugin@getAutoPlay
+   * @return {boolean} The value of autoPlay.
+   */
+  this.getAutoPlay = function()
+  {
+    return autoPlay;
+  };
+
+  /**
+   * [Required Function] Set the plugin id given by the Analytics Framework when
+   * this plugin is registered.
+   * @public
+   * @method IqPlugin#setPluginID
+   * @param  {string} newID The plugin id
+   */
+  this.setPluginID = function(newID)
+  {
+    id = newID;
+  };
+
+  /**
+   * [Required Function] Returns the stored plugin id, given by the Analytics Framework.
+   * @public
+   * @method IqPlugin#setPluginID
+   * @return  {string} The pluginID assigned to this instance from the Analytics Framework.
+   */
+  this.getPluginID = function()
+  {
+    return id;
+  };
+
+  /**
+   * [Required Function] Initialize the plugin with the given metadata.
+   * @public
+   * @method IqPlugin#init
+   */
+  this.init = function()
+  {
+    var missedEvents;
+    //if you need to process missed events, here is an example
+    if (_framework && OO._.isFunction(_framework.getRecordedEvents))
+    {
+      missedEvents = _framework.getRecordedEvents();
+    }
+    //use recorded events.
+
+    if (this.testMode)
+    {
+      trySetupAnalytics();
+    }
+    else if (!window.Ooyala)
+    {
+      OO.loadScriptOnce("https://analytics.ooyala.com/static/v3/analytics.js", trySetupAnalytics, sdkLoadError, SDK_LOAD_TIMEOUT);
```

Comment:
@aeng7  Should we be using just "//" with all urls?  Does the http version work?

Reply:
If http works, we should use //. Otherwise this is fine.

Reply:
I checked and "//analytics.ooyala.com/static/v3/analytics.js" seems to work here on my local. Changing it

Reply:
Fixed